In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings("ignore")

In [3]:
data=pd.read_csv("Myntra_Fasion_Clothing[1].csv")

In [4]:
pd.set_option("display.max_rows", None, "display.max_columns", None)

In [5]:
data.head()

URL  Product_id   BrandName  \
0  https://www.myntra.com/jeans/roadster/roadster...     2296012    Roadster   
1  https://www.myntra.com/track-pants/locomotive/...    13780156  LOCOMOTIVE   
2  https://www.myntra.com/shirts/roadster/roadste...    11895958    Roadster   
3  https://www.myntra.com/shapewear/zivame/zivame...     4335679      Zivame   
4  https://www.myntra.com/tshirts/roadster/roadst...    11690882    Roadster   

                Category Individual_category category_by_Gender  \
0            Bottom Wear               jeans                Men   
1            Bottom Wear         track-pants                Men   
2                Topwear              shirts                Men   
3  Lingerie & Sleep Wear           shapewear              Women   
4                Western             tshirts              Women   

                                         Description  DiscountPrice (in Rs)  \
0  roadster men navy blue slim fit mid rise clean...                  824.0   
1  locomotive men black  white solid slim fit tra...                  517.0   
2  roadster men navy white  black geometric print...                  629.0   
3  zivame women black saree shapewear zi3023core0...                  893.0   
4  roadster women white solid v neck pure cotton ...                    NaN   

   OriginalPrice (in Rs) DiscountOffer              SizeOption  Ratings  \
0                 1499.0       45% OFF      28, 30, 32, 34, 36      3.9   
1                 1149.0       55% OFF             S, M, L, XL      4.0   
2                 1399.0       55% OFF  38, 40, 42, 44, 46, 48      4.3   
3                 1295.0       31% OFF        S, M, L, XL, XXL      4.2   
4                  599.0       35% OFF         XS, S, M, L, XL      4.2   

   Reviews  
0    999.0  
1    999.0  
2    999.0  
3    999.0  
4    999.0

In [6]:
# first we change the column order as we put the url of the product at end after checking all details of product starting from product id
new_order = [1,2,3,4,5,6,7,8,9,10,11,12,0]
data=data[data.columns[new_order]]

In [49]:
data.head(50)

Product_id              BrandName                  Category  \
0      2296012               Roadster               Bottom Wear   
1     13780156             LOCOMOTIVE               Bottom Wear   
2     11895958               Roadster                   Topwear   
3      4335679                 Zivame     Lingerie & Sleep Wear   
4     11690882               Roadster                   Western   
5      2490950         Mast & Harbour                   Western   
6      6744434             HIGHLANDER               Bottom Wear   
7      8439415                  Mayra                   Western   
8     17381394               Roadster                   Western   
9      2359257               HERE&NOW               Bottom Wear   
10     7695793  HRX by Hrithik Roshan               Sports Wear   
11    10307375               Roadster                   Topwear   
12    12873874              Anubhutee               Indian Wear   
13    11634538                 Athena                   Western   
14     2312181               Roadster                   Western   
15    13842386             HIGHLANDER                   Topwear   
16    10473520                Vishudh               Indian Wear   
17    10561392                Sangria                 Plus Size   
18    12391750          Tokyo Talkies                   Western   
19     2522986             DressBerry                   Western   
20    17385142               Roadster                   Western   
21    12153330                  Anouk               Indian Wear   
22     1864573                 Enamor     Lingerie & Sleep Wear   
23    13205276          all about you                   Western   
24    11535928              KASSUALLY                   Western   
25     6552977                   RARE                   Western   
26     9476129                Zastraa                   Western   
27     2017530               WISSTLER                   Western   
28    11173948                Sangria                 Plus Size   
29    11118428               Roadster                   Western   
30    17383362               Roadster                   Western   
31     9971493                Vishudh                   Western   
32     7846797              Cottinfab                   Western   
33    10771696                 QUIERO                   Western   
34    10969948               Roadster                   Western   
35    11938872               ELEVANTO     Lingerie & Sleep Wear   
36    14154054          Tokyo Talkies                   Western   
37     2477061                  WROGN               Bottom Wear   
38    10187401                  WROGN               Bottom Wear   
39     6791064  HRX by Hrithik Roshan               Sports Wear   
40    12918912               Roadster                   Topwear   
41    11245584  HRX by Hrithik Roshan               Sports Wear   
42     8643285              SASSAFRAS                   Western   
43     9451341        her by invictus                   Western   
44     7526008             StyleStone                   Western   
45    11235516                  Mayra                   Western   
46    10064541               Roadster               Bottom Wear   
47      976816  HRX by Hrithik Roshan  Inner Wear &  Sleep Wear   
48    11269190            Moda Rapido                   Topwear   
49    10266495            Moda Rapido                   Topwear   

   Individual_category category_by_Gender  \
0                jeans                Men   
1          track-pants                Men   
2               shirts                Men   
3            shapewear              Women   
4              tshirts              Women   
5                 tops              Women   
6             trousers                Men   
7                 tops              Women   
8              tshirts              Women   
9                jeans                Men   
10              tights                Men   
11             tshirts                Men   
1

In [8]:
data.shape

(526564, 13)

In [9]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526564 entries, 0 to 526563
Data columns (total 13 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   Product_id             526564 non-null  int64  
 1   BrandName              526564 non-null  object 
 2   Category               526564 non-null  object 
 3   Individual_category    526564 non-null  object 
 4   category_by_Gender     526564 non-null  object 
 5   Description            526564 non-null  object 
 6   DiscountPrice (in Rs)  333406 non-null  float64
 7   OriginalPrice (in Rs)  526564 non-null  float64
 8   DiscountOffer          452258 non-null  object 
 9   SizeOption             526564 non-null  object 
 10  Ratings                190412 non-null  float64
 11  Reviews                190412 non-null  float64
 12  URL                    526564 non-null  object 
dtypes: float64(4), int64(1), object(8)
memory usage: 52.2+ MB


In [10]:
data.isnull().sum()

Product_id                    0
BrandName                     0
Category                      0
Individual_category           0
category_by_Gender            0
Description                   0
DiscountPrice (in Rs)    193158
OriginalPrice (in Rs)         0
DiscountOffer             74306
SizeOption                    0
Ratings                  336152
Reviews                  336152
URL                           0
dtype: int64

### there are 4 columns with  null values i.e.
1. DiscountPrice (in Rs)    193158
2. DiscountOffer             74306
3. Ratings                  336152
4. Reviews                  336152

In [11]:
data["DiscountPrice (in Rs)"].value_counts()

999.0      6341
599.0      5979
899.0      5740
799.0      4293
1199.0     4291
678.0      3404
699.0      3345
449.0      3215
719.0      3169
749.0      3044
649.0      2837
1499.0     2802
499.0      2752
1399.0     2590
679.0      2534
989.0      2374
1799.0     2337
839.0      2276
629.0      2189
1299.0     2014
779.0      2013
1049.0     2007
539.0      1986
399.0      1976
559.0      1887
479.0      1879
549.0      1849
1999.0     1821
1439.0     1768
1298.0     1725
959.0      1711
849.0      1700
1259.0     1684
1079.0     1631
349.0      1586
674.0      1577
1349.0     1565
1099.0     1552
1599.0     1499
879.0      1497
639.0      1495
909.0      1458
359.0      1438
2099.0     1410
1319.0     1369
659.0      1356
524.0      1341
2399.0     1304
494.0      1284
419.0      1268
1249.0     1256
675.0      1244
1039.0     1238
519.0      1214
1119.0     1166
1749.0     1137
949.0      1134
791.0      1111
759.0      1104
979.0      1091
1679.0     1091
809.0      1039
584.0   

If we observe this, we can understand that there are a lot of single values in data["DiscountPrice (in Rs)"] which 
can unnecessarily affect the model if we try to train it using this data.

In [12]:
data['DiscountPrice (in Rs)'].describe()

count    333406.000000
mean       1237.444401
std        1052.063517
min         127.000000
25%         659.000000
50%         952.000000
75%        1469.000000
max       27996.000000
Name: DiscountPrice (in Rs), dtype: float64

In [13]:
# we can put or remove null value according to different price of different category of products
data['Category'].value_counts()

Indian Wear                 145845
Western                     140992
Topwear                      74537
Bottom Wear                  55439
Lingerie & Sleep Wear        55258
Sports Wear                  20627
Inner Wear &  Sleep Wear     20370
Plus Size                    13496
Name: Category, dtype: int64

In [14]:

data.groupby('Category').mean()['DiscountPrice (in Rs)']

Category
Bottom Wear                 1351.207867
Indian Wear                 1493.337491
Inner Wear &  Sleep Wear     691.839761
Lingerie & Sleep Wear        811.902842
Plus Size                    923.694598
Sports Wear                 1097.378374
Topwear                     1138.634061
Western                     1790.085106
Name: DiscountPrice (in Rs), dtype: float64

In [15]:
#in the following method, we have used groupby() and transform() functions to replace Null values.
data['DiscountPrice (in Rs)'] = data['DiscountPrice (in Rs)'].fillna(data.groupby('Category')['DiscountPrice (in Rs)'].transform('mean'))

In [16]:
data['DiscountPrice (in Rs)'].describe() 
# now we can see that the mean value changes

count    526564.000000
mean       1374.993685
std         878.604268
min         127.000000
25%         798.000000
50%        1299.000000
75%        1790.085106
max       27996.000000
Name: DiscountPrice (in Rs), dtype: float64

In [17]:
data['DiscountPrice (in Rs)'].isnull().sum() # now we succefully remove the null values of this column

0

In [18]:
data['Ratings'].value_counts()

4.3    22529
4.2    20573
4.4    18339
4.1    17278
4.0    17203
4.5    13295
3.9    11358
3.8    11034
4.6     9668
3.7     7042
4.7     5823
3.6     5546
4.8     4938
3.5     4135
3.4     3382
5.0     3235
3.3     2983
3.0     2523
3.2     1810
4.9     1570
3.1     1053
2.8      940
2.9      569
2.7      557
2.5      533
2.6      467
2.0      441
2.3      405
1.0      341
2.4      259
2.2      154
1.5      101
1.8       92
1.7       65
2.1       54
1.6       39
1.3       35
1.9       23
1.4       11
1.2        7
1.1        2
Name: Ratings, dtype: int64

In [19]:
data['Ratings'].describe() # we can't use mean value in null values here because mean is close to max value which is not a good odea to use in this 

count    190412.000000
mean          4.094892
std           0.494158
min           1.000000
25%           3.900000
50%           4.200000
75%           4.400000
max           5.000000
Name: Ratings, dtype: float64

In [20]:
data.columns

Index(['Product_id', 'BrandName', 'Category', 'Individual_category',
       'category_by_Gender', 'Description', 'DiscountPrice (in Rs)',
       'OriginalPrice (in Rs)', 'DiscountOffer', 'SizeOption', 'Ratings',
       'Reviews', 'URL'],
      dtype='object')

In [21]:
data.groupby('Category').mean()['Ratings'] # we can't use mean according to each category so we have to drop this idea too


Category
Bottom Wear                 4.034112
Indian Wear                 4.006331
Inner Wear &  Sleep Wear    4.186591
Lingerie & Sleep Wear       4.131093
Plus Size                   4.127853
Sports Wear                 4.198555
Topwear                     4.130325
Western                     4.099510
Name: Ratings, dtype: float64

In [22]:
data.groupby('Individual_category').mean()['Ratings']

Individual_category
baby-dolls              4.004233
bath-robe               4.074129
blazers                 3.752813
boots                   3.650000
boxers                  4.076269
bra                     4.101788
bracelet                     NaN
briefs                  4.226426
burqas                  4.009524
camisoles               4.237810
capris                  4.215174
casual-shoes            4.037500
churidar                4.118692
clothing-set            4.190000
co-ords                 4.016346
coats                   3.926070
dhotis                  4.017105
dress-material          3.933333
dresses                 4.035871
dungarees               4.083333
dupatta                 4.190800
earrings                4.281667
flats                   3.992593
flip-flops                   NaN
hair-accessory          3.700000
handbags                     NaN
harem-pants             4.250000
heels                   4.200000
innerwear-vests         4.139825
jackets                

In [25]:
data['Individual_category'].value_counts()

tshirts                 61198
sarees                  57915
tops                    39126
kurtas                  38984
dresses                 35590
shirts                  32692
kurta-sets              26592
jeans                   25206
trousers                24706
bra                     18097
track-pants             16407
shorts                  13580
sweatshirts             13124
night-suits             11527
jackets                 10625
briefs                   9234
lounge-pants             8056
sweaters                 6732
nightdress               6465
leggings                 6005
skirts                   4622
palazzos                 3805
tights                   3800
socks                    3705
dupatta                  3536
trunk                    3081
jumpsuit                 3039
boxers                   3036
lehenga-choli            3012
nehru-jackets            2916
lounge-shorts            2634
innerwear-vests          1885
dress-material           1864
camisoles 

In [27]:
# if we remove the data that is less than 10 in counts there is no efect on data
counts = data['Individual_category'].value_counts()

data = data[~data['Individual_category'].isin(counts[counts < 11].index)]

In [30]:
data['Individual_category'].value_counts()

tshirts                 61198
sarees                  57915
tops                    39126
kurtas                  38984
dresses                 35590
shirts                  32692
kurta-sets              26592
jeans                   25206
trousers                24706
bra                     18097
track-pants             16407
shorts                  13580
sweatshirts             13124
night-suits             11527
jackets                 10625
briefs                   9234
lounge-pants             8056
sweaters                 6732
nightdress               6465
leggings                 6005
skirts                   4622
palazzos                 3805
tights                   3800
socks                    3705
dupatta                  3536
trunk                    3081
jumpsuit                 3039
boxers                   3036
lehenga-choli            3012
nehru-jackets            2916
lounge-shorts            2634
innerwear-vests          1885
dress-material           1864
camisoles 

In [31]:
data.groupby('Individual_category').mean()['Ratings'] 

Individual_category
baby-dolls              4.004233
bath-robe               4.074129
blazers                 3.752813
boxers                  4.076269
bra                     4.101788
briefs                  4.226426
burqas                  4.009524
camisoles               4.237810
capris                  4.215174
casual-shoes            4.037500
churidar                4.118692
clothing-set            4.190000
co-ords                 4.016346
coats                   3.926070
dhotis                  4.017105
dress-material          3.933333
dresses                 4.035871
dungarees               4.083333
dupatta                 4.190800
earrings                4.281667
flats                   3.992593
innerwear-vests         4.139825
jackets                 4.114520
jeans                   3.979808
jeggings                4.070702
jumpsuit                4.040460
kurta-sets              3.994493
kurtas                  4.044259
kurtis                  4.115789
leggings               

In [33]:
data["Ratings"].describe()

count    190384.000000
mean          4.094888
std           0.494151
min           1.000000
25%           3.900000
50%           4.200000
75%           4.400000
max           5.000000
Name: Ratings, dtype: float64

In [34]:
data['Ratings'] = data['Ratings'].fillna(data.groupby('Individual_category')['Ratings'].transform('mean'))

In [35]:
data["Ratings"].describe() #so after filling the values there is not much change after cleaning

count    526506.000000
mean          4.078776
std           0.310195
min           1.000000
25%           3.994493
50%           4.066997
75%           4.207535
max           5.000000
Name: Ratings, dtype: float64

In [41]:
data.isnull().sum()

Product_id                    0
BrandName                     0
Category                      0
Individual_category           0
category_by_Gender            0
Description                   0
DiscountPrice (in Rs)         0
OriginalPrice (in Rs)         0
DiscountOffer             74297
SizeOption                    0
Ratings                       0
Reviews                  336122
URL                           0
dtype: int64

In [48]:
data["Reviews"].value_counts() #here we can see there are lot of reviews where the value counts are less then 15 


5.0      12168
6.0       9876
7.0       8326
8.0       7345
9.0       6372
10.0      5696
11.0      5111
12.0      4553
0.0       4454
13.0      4255
14.0      3881
4.0       3707
3.0       3620
15.0      3554
16.0      3297
17.0      3011
2.0       2831
18.0      2794
19.0      2714
20.0      2591
21.0      2323
22.0      2297
23.0      2162
24.0      1993
25.0      1927
26.0      1756
28.0      1722
27.0      1697
29.0      1602
30.0      1558
31.0      1502
32.0      1394
34.0      1334
33.0      1311
35.0      1246
36.0      1225
37.0      1159
38.0      1152
39.0      1097
40.0      1062
43.0      1033
41.0      1016
42.0      1005
45.0       952
44.0       939
46.0       885
47.0       864
50.0       838
49.0       805
48.0       793
52.0       777
51.0       752
53.0       715
54.0       711
55.0       699
56.0       641
57.0       632
59.0       612
62.0       606
61.0       602
58.0       600
60.0       582
64.0       549
66.0       538
65.0       538
63.0       526
67.0      

In [39]:
data["Reviews"].describe()

count    190384.000000
mean         61.991402
std         125.719642
min           0.000000
25%           8.000000
50%          18.000000
75%          52.000000
max         999.000000
Name: Reviews, dtype: float64

In [52]:
# if we remove the data that is less than 15 in counts there is no efect on data
#counts = data['Reviews'].value_counts()

#+data = data[~data['Reviews'].isin(counts[counts < 16].index)]

In [53]:
#data["Reviews"].describe()

count    186458.000000
mean         48.352026
std          81.589448
min           0.000000
25%           8.000000
50%          18.000000
75%          49.000000
max         789.000000
Name: Reviews, dtype: float64